AArch64 toolchain: cortex-A53 will fail if using GCC14.1 and glibc

### Describe the bug

Cortex-a53 target will fail if using **gcc14.1 and glibc** ( libatomic will fail )

Logs below:

```
make[8]: Entering directory '/media/zhou/u2nvme/dev/openwrt/build_dir/toolchain-aarch64_cortex-a53_gcc-14.1.0_glibc/gcc-14.1.0-final/aarch64-openwrt-linux-gnu/libatomic'
/bin/bash ./libtool  --tag=CC   --mode=compile /media/zhou/u2nvme/dev/openwrt/build_dir/toolchain-aarch64_cortex-a53_gcc-14.1.0_glibc/gcc-14.1.0-final/./gcc/xgcc -B/media/zhou/u2nvme/dev/openwrt/build_dir/toolchain-aarch64_cortex-a53_gcc-14.1.0_glibc/gcc-14.1.0-final/./gcc/ -B/media/zhou/u2nvme/dev/openwrt/staging_dir/toolchain-aarch64_cortex-a53_gcc-14.1.0_glibc/aarch64-openwrt-linux-gnu/bin/ -B/media/zhou/u2nvme/dev/openwrt/staging_dir/toolchain-aarch64_cortex-a53_gcc-14.1.0_glibc/aarch64-openwrt-linux-gnu/lib/ -isystem /media/zhou/u2nvme/dev/openwrt/staging_dir/toolchain-aarch64_cortex-a53_gcc-14.1.0_glibc/aarch64-openwrt-linux-gnu/include -isystem /media/zhou/u2nvme/dev/openwrt/staging_dir/toolchain-aarch64_cortex-a53_gcc-14.1.0_glibc/aarch64-openwrt-linux-gnu/sys-include    -DHAVE_CONFIG_H -I/media/zhou/u2nvme/dev/openwrt/build_dir/toolchain-aarch64_cortex-a53_gcc-14.1.0_glibc/gcc-14.1.0/libatomic/config/aarch64 -I/media/zhou/u2nvme/dev/openwrt/build_dir/toolchain-aarch64_cortex-a53_gcc-14.1.0_glibc/gcc-14.1.0/libatomic/config/linux/aarch64 -I/media/zhou/u2nvme/dev/openwrt/build_dir/toolchain-aarch64_cortex-a53_gcc-14.1.0_glibc/gcc-14.1.0/libatomic/config/posix -I/media/zhou/u2nvme/dev/openwrt/build_dir/toolchain-aarch64_cortex-a53_gcc-14.1.0_glibc/gcc-14.1.0/libatomic -I.    -mno-outline-atomics -Wall -Werror   -pthread -Os -pipe -mcpu=cortex-a53 -fno-caller-saves -fno-plt -fhonour-copts -Wformat -Werror=format-security -fstack-protector -D_FORTIFY_SOURCE=1 -Wl,-z,now -Wl,-z,relro -g -Os -MT load_1_1_.lo -MD -MP -MF .deps/load_1_1_.lo.Ppo -DN=1 -DIFUNC_ALT=1 -march=armv8-a+lse -c -o load_1_1_.lo /media/zhou/u2nvme/dev/openwrt/build_dir/toolchain-aarch64_cortex-a53_gcc-14.1.0_glibc/gcc-14.1.0/libatomic/load_n.c
libtool: compile:  /media/zhou/u2nvme/dev/openwrt/build_dir/toolchain-aarch64_cortex-a53_gcc-14.1.0_glibc/gcc-14.1.0-final/./gcc/xgcc -B/media/zhou/u2nvme/dev/openwrt/build_dir/toolchain-aarch64_cortex-a53_gcc-14.1.0_glibc/gcc-14.1.0-final/./gcc/ -B/media/zhou/u2nvme/dev/openwrt/staging_dir/toolchain-aarch64_cortex-a53_gcc-14.1.0_glibc/aarch64-openwrt-linux-gnu/bin/ -B/media/zhou/u2nvme/dev/openwrt/staging_dir/toolchain-aarch64_cortex-a53_gcc-14.1.0_glibc/aarch64-openwrt-linux-gnu/lib/ -isystem /media/zhou/u2nvme/dev/openwrt/staging_dir/toolchain-aarch64_cortex-a53_gcc-14.1.0_glibc/aarch64-openwrt-linux-gnu/include -isystem /media/zhou/u2nvme/dev/openwrt/staging_dir/toolchain-aarch64_cortex-a53_gcc-14.1.0_glibc/aarch64-openwrt-linux-gnu/sys-include -DHAVE_CONFIG_H -I/media/zhou/u2nvme/dev/openwrt/build_dir/toolchain-aarch64_cortex-a53_gcc-14.1.0_glibc/gcc-14.1.0/libatomic/config/aarch64 -I/media/zhou/u2nvme/dev/openwrt/build_dir/toolchain-aarch64_cortex-a53_gcc-14.1.0_glibc/gcc-14.1.0/libatomic/config/linux/aarch64 -I/media/zhou/u2nvme/dev/openwrt/build_dir/toolchain-aarch64_cortex-a53_gcc-14.1.0_glibc/gcc-14.1.0/libatomic/config/posix -I/media/zhou/u2nvme/dev/openwrt/build_dir/toolchain-aarch64_cortex-a53_gcc-14.1.0_glibc/gcc-14.1.0/libatomic -I. -mno-outline-atomics -Wall -Werror -pthread -Os -pipe -mcpu=cortex-a53 -fno-caller-saves -fno-plt -fhonour-copts -Wformat -Werror=format-security -fstack-protector -D_FORTIFY_SOURCE=1 -Wl,-z,now -Wl,-z,relro -g -Os -MT load_1_1_.lo -MD -MP -MF .deps/load_1_1_.lo.Ppo -DN=1 -DIFUNC_ALT=1 -march=armv8-a+lse -c /media/zhou/u2nvme/dev/openwrt/build_dir/toolchain-aarch64_cortex-a53_gcc-14.1.0_glibc/gcc-14.1.0/libatomic/load_n.c  -fPIC -DPIC -o .libs/load_1_1_.o
cc1: error: switch '-mcpu=cortex-a53' conflicts with '-march=armv8-a+lse' switch and resulted in options '+lse+nocrc' being added [-Werror]
cc1: all warnings being treated as errors
Makefile:899: recipe for target 'load_1_1_.lo' failed
make[8]: *** [load_1_1_.lo] Error 1
make[8]: Leaving directory '/media/zhou/u2nvme/dev/openwrt/build_dir/toolchain-aarch64_cortex-a53_gcc-14.1.0_glibc/gcc-14.1.0-final/aarch64-openwrt-linux-gnu/libatomic'
Makefile:650: recipe for target 'all-recursive' failed
make[7]: *** [all-recursive] Error 1
make[7]: Leaving directory '/media/zhou/u2nvme/dev/openwrt/build_dir/toolchain-aarch64_cortex-a53_gcc-14.1.0_glibc/gcc-14.1.0-final/aarch64-openwrt-linux-gnu/libatomic'
Makefile:468: recipe for target 'all' failed
make[6]: *** [all] Error 2
make[6]: Leaving directory '/media/zhou/u2nvme/dev/openwrt/build_dir/toolchain-aarch64_cortex-a53_gcc-14.1.0_glibc/gcc-14.1.0-final/aarch64-openwrt-linux-gnu/libatomic'
Makefile:16994: recipe for target 'all-target-libatomic' failed
make[5]: *** [all-target-libatomic] Error 2
make[5]: Leaving directory '/media/zhou/u2nvme/dev/openwrt/build_dir/toolchain-aarch64_cortex-a53_gcc-14.1.0_glibc/gcc-14.1.0-final'
Makefile:1062: recipe for target 'all' failed
make[4]: *** [all] Error 2
make[4]: Leaving directory '/media/zhou/u2nvme/dev/openwrt/build_dir/toolchain-aarch64_cortex-a53_gcc-14.1.0_glibc/gcc-14.1.0-final'
Makefile:90: recipe for target '/media/zhou/u2nvme/dev/openwrt/build_dir/toolchain-aarch64_cortex-a53_gcc-14.1.0_glibc/gcc-14.1.0-final/.built' failed
make[3]: *** [/media/zhou/u2nvme/dev/openwrt/build_dir/toolchain-aarch64_cortex-a53_gcc-14.1.0_glibc/gcc-14.1.0-final/.built] Error 2
make[3]: Leaving directory '/media/zhou/u2nvme/dev/openwrt/toolchain/gcc/final'
time: toolchain/gcc/final/compile#291.94#60.80#332.20
    ERROR: toolchain/gcc/final failed to build.
toolchain/Makefile:91: recipe for target 'toolchain/gcc/final/compile' failed
make[2]: *** [toolchain/gcc/final/compile] Error 1
make[2]: Leaving directory '/media/zhou/u2nvme/dev/openwrt'
toolchain/Makefile:89: recipe for target '/media/zhou/u2nvme/dev/openwrt/staging_dir/toolchain-aarch64_cortex-a53_gcc-14.1.0_glibc/stamp/.toolchain_compile' failed
make[1]: *** [/media/zhou/u2nvme/dev/openwrt/staging_dir/toolchain-aarch64_cortex-a53_gcc-14.1.0_glibc/stamp/.toolchain_compile] Error 2
make[1]: Leaving directory '/media/zhou/u2nvme/dev/openwrt'
/media/zhou/u2nvme/dev/openwrt/include/toplevel.mk:246: recipe for target 'world' failed
make: *** [world] Error 2
```
config attached here 
[fail_config.zip](https://github.com/user-attachments/files/16492296/fail_config.zip)


the main reason is 
`cc1: error: switch '-mcpu=cortex-a53' conflicts with '-march=armv8-a+lse' switch and resulted in options '+lse+nocrc' being added [-Werror]`

regarding to #15414
and c758d6427c40e7518f0a85ce607930335c88e56f
No idea why -mno-outline-atomics did not passed to the toolchain properly

### OpenWrt version

r27076-d604b2699b

### OpenWrt release

SNAPSHOT

### OpenWrt target/subtarget

mediatek/filogic

### Device

gl-mt3000

### Image kind

Self-built image

### Steps to reproduce

use configs above and using gcc14 with glibc , then toolchain/gcc/final/compile will fail


### Actual behaviour

cc1: error: switch '-mcpu=cortex-a53' conflicts with '-march=armv8-a+lse' switch and resulted in options '+lse+nocrc' being added [-Werror]

### Expected behaviour

toolchain/gcc/final/compile should deal with -mcpu and -march properly

### Additional info

_No response_

### Diffconfig

_No response_

### Terms

- [X] I am reporting an issue for OpenWrt, not an unsupported fork.

--- a/toolchain/gcc/patches-14.x/830-aarch64-libatomic.patch
+++ b/toolchain/gcc/patches-14.x/830-aarch64-libatomic.patch
@@ -1,0 +1,32 @@
+libatomic: Do not enforce march on aarch64
+
+Inspired by The Yocto Project [1].
+
+[1] https://github.com/yoctoproject/poky/blob/51192a79f1717786dda42776f916c3d97ada7971/meta/recipes-devtools/gcc/gcc/0022-libatomic-Do-not-enforce-march-on-aarch64.patch
+
+Signed-off-by: Konstantin Demin <rockdrilla@gmail.com>
+
+ libatomic/Makefile.am | 1 -
+ libatomic/Makefile.in | 1 -
+ 2 files changed, 2 deletions(-)
+
+--- a/libatomic/Makefile.am
++++ b/libatomic/Makefile.am
+@@ -130,7 +130,6 @@ libatomic_la_LIBADD = $(foreach s,$(SIZES),$(addsuffix _$(s)_.lo,$(SIZEOBJS)))
+ ## On a target-specific basis, include alternates to be selected by IFUNC.
+ if HAVE_IFUNC
+ if ARCH_AARCH64_LINUX
+-IFUNC_OPTIONS	     = -march=armv8-a+lse
+ libatomic_la_LIBADD += $(foreach s,$(SIZES),$(addsuffix _$(s)_1_.lo,$(SIZEOBJS)))
+ 
+ endif
+--- a/libatomic/Makefile.in
++++ b/libatomic/Makefile.in
+@@ -452,7 +452,6 @@ M_SRC = $(firstword $(filter %/$(M_FILE), $(all_c_files)))
+ libatomic_la_LIBADD = $(foreach s,$(SIZES),$(addsuffix \
+ 	_$(s)_.lo,$(SIZEOBJS))) $(am__append_1) $(am__append_2) \
+ 	$(am__append_3) $(am__append_4)
+-@ARCH_AARCH64_LINUX_TRUE@@HAVE_IFUNC_TRUE@IFUNC_OPTIONS = -march=armv8-a+lse
+ @ARCH_ARM_LINUX_TRUE@@HAVE_IFUNC_TRUE@IFUNC_OPTIONS = -march=armv7-a+fp -DHAVE_KERNEL64
+ @ARCH_I386_TRUE@@HAVE_IFUNC_TRUE@IFUNC_OPTIONS = -march=i586
+ @ARCH_X86_64_TRUE@@HAVE_IFUNC_TRUE@IFUNC_OPTIONS = -mcx16 -mcx16

